api/clients/eigenda_client.go

```diff
@@ -251,10 +297,21 @@ func (m *EigenDAClient) putBlob(ctx context.Context, rawData []byte, resultChan
 					alreadyWaitingForDispersal = true
 				}
 			case grpcdisperser.BlobStatus_FAILED:
-				errChan <- fmt.Errorf("EigenDA blob dispersal failed in processing, requestID=%s: %w", base64RequestID, err)
+				// This can happen for a few reasons:
+				// 1. blob has expired, a client retrieve after 14 days. Sounds like 400 errors, but not sure this can happen during dispersal...
+				// 2. internal logic error while requesting encoding (shouldn't happen), but should probably return api.ErrorFailover
+				// 3. wait for blob finalization from confirmation and blob retry has exceeded its limit.
+				//    Probably from a chain re-org. See https://github.com/Layr-Labs/eigenda/blob/master/disperser/batcher/finalizer.go#L179-L189.
+				//    So we should be returning 500 to force a blob resubmission (not eigenda's fault but until
```

Comment:
In v2, disperser likely will not return a failed state, because there is no onchain transaction. Then the source to enter into a failed state most is mostly because of insufficient signature. In that case, the disperser will keep retry, and it would not return a failed state. Failed state probably just is going to be some catastrophic event on disperser.

Reply:
Think we shouldn't add comments about v2 until its actually implemented though? Do you want to just add this note to the notion doc we started yesterday?